### PR TITLE
Update issue templates

### PR DESCRIPTION
**Pull request by mwcraig**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/add-maintainer-checklist.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+name: Add maintainer checklist
+about: Reminder to check list of steps when modifying team
+title: ''
+labels: ''
+assignees: ''
+
+---
+
+If you are adding or removing someone from a team please check that the list of items here has been completed: https://github.com/astropy/astropy-project/blob/main/policies/adding-new-role-members.md
```
Comment (kelle) [inline .github/ISSUE_TEMPLATE/add-maintainer-checklist.md:10]:
```suggestion
If you are adding or removing someone from a team please check that the list of items on the [Adding New Role Members page](https://github.com/astropy/astropy-project/blob/main/policies/adding-new-role-members.md) has been completed. In particular,

- [ ] PR to the Team page
- [ ] email to the community
- [ ] privileges and/or permissions updated 
- [ ] added to mailing list

```

**Review by kelle — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/add-maintainer-checklist.md (hunk 1 above)
Comment (hamogu) [reply .github/ISSUE_TEMPLATE/add-maintainer-checklist.md:10]: I can't suggest suggestions for a suggestion, but I suggest here "added to astropy-core-maintainers mailing list".

File: .github/ISSUE_TEMPLATE/add-maintainer-checklist.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+assignees: ''
+
+---
+
+If you are adding or removing someone from a team please check that the list of items on the [Adding New Role Members page](https://github.com/astropy/astropy-project/blob/main/policies/adding-new-role-members.md) has been completed. In particular,
+
+- [ ] PR to the Team page
+- [ ] email to the community
+- [ ] privileges and/or permissions updated 
+- [ ] added to core maintainer mailing list
```
Comment (kelle) [inline .github/ISSUE_TEMPLATE/add-maintainer-checklist.md:15]:
```suggestion
- [ ] added to astropy-core-maintainers mailing list
```

**Review by kelle — APPROVED** [review]

**Review by eteq — APPROVED** [review]
I think this is great.  It might need to get some minor updates the first time someone actually uses it, but that's fine!

File: .github/ISSUE_TEMPLATE/add-maintainer-checklist.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+assignees: ''
+
+---
+
+If you are adding or removing someone from a team please check that the list of items on the [Adding New Role Members page](https://github.com/astropy/astropy-project/blob/main/policies/adding-new-role-members.md) has been completed. In particular,
+
+- [ ] PR to the Team page
+- [ ] email to the community
+- [ ] privileges and/or permissions updated 
+- [ ] added to astropy-core-maintainers mailing list
```
Comment (adrn) [inline .github/ISSUE_TEMPLATE/add-maintainer-checklist.md:15]:
```suggestion
- [ ] privileges and/or permissions updated 
- [ ] (if adding a new member) email to the community
- [ ] (if adding a new member) add to astropy-core-maintainers mailing list
```

**Pull request merged**